src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -121,19 +129,33 @@ function useChatUtilsHook(): UseChatUtilsHook {
     setChannelAsRead(channel);
     if (channel?.channelType === ANON_PM) {
       if (from === GROUP_INFO) {
-        return openChat('SampleChatScreen');
+        return openChat('SampleChatScreen', 'AnonymousChannelList');
       }
       navigation.navigate('SampleChatScreen');
     } else {
       if (from === GROUP_INFO) {
-        return openChat('SignedChatScreen');
+        return openChat('SignedChatScreen', 'SignedChannelList');
       }
       navigation.navigate('SignedChatScreen');
     }
 
     return null;
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `goToChatScreen` function has been updated to handle navigation based on the `from` parameter. If `from` equals 'GROUP_INFO', it navigates to either 'SampleChatScreen' or 'SignedChatScreen' depending on the channel type. If `from` does not equal 'GROUP_INFO', it directly navigates to either 'SampleChatScreen' or 'SignedChatScreen' without calling `openChat`. This change seems fine as it provides more control over navigation based on where the function is called from.



<!-- This is an auto-generated comment by OSS CodeRabbit -->